spyder/widgets/mixins.py

```diff
@@ -64,72 +70,260 @@ def calculate_real_position(self, point):
 
         # Saving cursor position:
         if at_position is None:
             at_position = self.get_position('cursor')
-        self.calltip_position = at_position
 
-        # Preparing text:
-        if signature:
-            text, wrapped_textlines = self._format_signature(text)
-        else:
-            if isinstance(text, list):
-                text = "\n    ".join(text)
-            text = text.replace('\n', '<br>')
-            if len(text) > self.calltip_size:
-                text = text[:self.calltip_size] + " ..."
+        # FIXME: What id this used for?
+        self.calltip_position = at_position
 
-        # Formatting text
-        font = self.font()
-        size = font.pointSize()
-        family = font.family()
-        format1 = '<div style=\'font-family: "%s"; font-size: %spt; color: %s\'>'\
-                  % (family, size, color)
-        format2 = '<div style=\'font-family: "%s"; font-size: %spt\'>'\
-                  % (family, size-1 if size > 9 else size)
-        tiptext = format1 + ('<b>%s</b></div>' % title) + '<hr>' + \
-                  format2 + text + "</div>"
-
-        # Showing tooltip at cursor position:
-        cx, cy = self.get_coordinates('cursor')
         if at_point is not None:
+            # Showing tooltip at point position
             cx, cy = at_point.x(), at_point.y()
-        if at_line is not None:
+        elif at_line is not None:
+            # Showing tooltip at line
             cx = 5
             cursor = QTextCursor(self.document().findBlockByNumber(at_line-1))
             cy = self.cursorRect(cursor).top()
+        else:
+            # Showing tooltip at cursor position
+            # FIXME: why is at_position not being used to calculate the
+            # coordinates?
+            cx, cy = self.get_coordinates('cursor')
+
+        # Calculate vertical delta
+        font = self.font()
+        delta = font.pointSize() + 5
+
+        # Map to global coordinates
         point = self.mapToGlobal(QPoint(cx, cy))
         point = self.calculate_real_position(point)
-        point.setY(point.y()+font.pointSize()+5)
-        if signature:
-            self.calltip_widget.show_tip(point, tiptext, wrapped_textlines)
-        else:
-            QToolTip.showText(point, tiptext)
+        point.setY(point.y() + delta)
+
+        return point
+
+    def _update_stylesheet(self, widget):
+        """Update the background stylesheet to make it lighter."""
+        if is_dark_interface():
+            css = qdarkstyle.load_stylesheet_from_environment()
+            widget.setStyleSheet(css)
+            palette = widget.palette()
+            background = palette.color(palette.Window).lighter().name()
+            name = widget.__class__.__name__
+            widget.setObjectName(name)
+            extra_css = '''
+                {0}#{0} {{background-color:{1};}}'''.format(name, background)
+            widget.setStyleSheet(css + extra_css)

```

Comment:
So I could not get it to work by updating the palette, but the palette colors are updated by the stylesheet that is why I first apply the CSS then get the color to make it lighter and then reapply an extended CSS which includes the lighter background.

Reply:
Ok, no problem with that. Hopefully @dpizetta will release `qdarkstyle` 3 soon, making this far more easier.